cmd/launcher/flare.go

```diff
@@ -11,23 +12,45 @@ import (
 	"github.com/kolide/kit/ulid"
 	"github.com/kolide/launcher/pkg/agent/flags"
 	"github.com/kolide/launcher/pkg/agent/knapsack"
+	"github.com/kolide/launcher/pkg/agent/types"
 	"github.com/kolide/launcher/pkg/debug/checkups"
+	"github.com/kolide/launcher/pkg/debug/flareshipping"
 	"github.com/kolide/launcher/pkg/launcher"
 )
 
+// runFlareInt is an empty struct that implements the RunFlare method so we can pass to flare shipper
+type runFlareInt struct{}
+
+func (_ runFlareInt) RunFlare(ctx context.Context, k types.Knapsack, flare io.Writer, environment checkups.RuntimeEnvironmentType) error {
+	return checkups.RunFlare(ctx, k, flare, environment)
+}
+
 func runFlare(args []string) error {
 	// Flare assumes a launcher installation (at least partially) exists
 	// Overriding some of the default values allows options to be parsed making this assumption
 	// TODO this stuff needs some deeper thinking
 	launcher.DefaultAutoupdate = true
-	setDefaultPaths()
+	launcher.SetDefaultPaths()
 	_ = launcher.DefaultBinDirectoryPath
 
 	opts, err := launcher.ParseOptions("flare", args)
 	if err != nil {
 		return err
 	}
 
+	logger := log.NewLogfmtLogger(os.Stdout)
+	fcOpts := []flags.Option{flags.WithCmdLineOpts(opts)}
+	flagController := flags.NewFlagController(logger, nil, fcOpts...)
+	k := knapsack.New(nil, flagController, nil)
+
+	ctx := context.Background()
+
+	var requestUrl = env.String("KOLIDE_AGENT_FLARE_REQUEST_URL", "")
+	if requestUrl != "" {
+		return flareshipping.RunFlareShip(logger, k, runFlareInt{}, requestUrl)
+	}
```

Comment:
This feels a little off. What if there was a `flareshipping.Shipper` that had an `io.Writer` and could be passed in on line 70?

(Might need to change that to using an io.WriteCloser

Reply:
I took the suggestion a little further and just made a "shipper" instead of a "flare shipper" it still just functional though, did not introduce a new struct. Please take a look and let me know what you thing.